Marlin/src/module/probe.cpp

```diff
@@ -516,6 +541,10 @@ float Probe::run_z_probe(const bool sanity_check/*=true*/) {
   #if TOTAL_PROBING == 2
 
     // Do a first probe at the fast speed
+    #if ENABLED(PROBE_NEEDS_TARE)
+      if(tare_z_probe()) return true;
```

Comment:
This should probably return `NAN` instead of `true`, to match what happens if probing fails a few lines down.